relay-auth/src/lib.rs

```diff
@@ -687,6 +698,147 @@ impl RegisterResponse {
     }
 }
 
+/// Signature within relay which can be valid or invalid.
+///
+/// Invalid does not refer to a failed verification but rather to a signature
+/// that is not well formatted.
+#[derive(Debug, Clone, PartialEq)]
+pub enum RelaySignature {
+    /// The signature has a valid format. It does not mean that it is verified.
+    Valid(RelaySignatureData),
+    /// The signature could not be converted into the proper signature format.
+    Invalid(SignatureError),
+}
+
+impl From<SignatureError> for RelaySignature {
+    fn from(value: SignatureError) -> Self {
+        Self::Invalid(value)
+    }
+}
+
+/// Represents whether credentials are required for signing a request.
+pub enum TrySign {
```

Comment:
I think this makes more sense if it was part of the upstream, where the `Request` is defined, it's not really related to the auth itself.